conda/core/subdir_data.py

```diff
@@ -235,6 +240,27 @@ def iter_records(self):
             self.load()
         return iter(self._package_records)
 
+    def _load_state(self) -> dict:
+        """
+        Cache headers and additional data needed to keep track of the cache are
+        stored separately, instead of the previous "added to repodata.json"
+        arrangement.
+        """
+        try:
+            state_path = pathlib.Path(self.cache_path_state)
+            log.debug("Load %s cache from %s", self.repodata_fn, state_path)
+            # efficient according to scalene profiler; about equal to
+            # json.loads(state_path.read_text()) and better that open("rb")
+            with state_path.open("r") as s:
+                state = json.load(s)
+            return state
+        except (json.JSONDecodeError, OSError):
+            log.debug("Could not load state", exc_info=True)
+            return {}
+
+    def _save_state(self, state: dict):
+        return pathlib.Path(self.cache_path_state).write_text(json.dumps(state, indent=True))
```

Comment:
Do we need to do any exception handling for this, so the API works the same as with `_load_state`?

Reply:
Alternatively, you could move the error handling out of `_load_state` into where it's called.

Reply:
If you can't save this, it's as bad as not being able to save repodata.json

Reply:
Okay.